### ISSUE TITLE
Feature: Condense/exclude time on the x-axis for given time windows

### ISSUE THREAD

**Issue by timgriffiths** [grafana/grafana#2418]
We are currently investigating graphing stock market data using grafana, and as such it only has certain periods where we actually receive data to graph. it would be really good if we could define windows in which we expect to see data and then the graph just collapses the x-axis up so it just show 1 continuous line with a dashed line (or something) where the windows starts and stops

eg 
just show data between 9am - 3pm && 6pm - 9pm

**Comment by torkelo** [comment]
not sure I understand, do you want to reduce the x axis so there is no gap between 3pm and 6pm, but if there this gap is collapsed how is a dashed line supposed to be drawn in the missing region?

**Comment by timgriffiths** [comment]
So something like this

![image](https://cloud.githubusercontent.com/assets/4284863/9053157/5b93f054-3ab4-11e5-9ef0-e8875b417c3a.png)

**Comment by timgriffiths** [comment]
Do you require any more detail @torkelo?

**Comment by torkelo** [comment]
no, I understand

**Comment by LaurentCognet** [comment]
Hello, It would be very convenient to add this feature for us too !
Hope it will come !
Congratulation for the job already done !

**Comment by sauravkedia10** [comment]
Hi, Have you been able to work on this issue?

**Comment by LaurentCognet** [comment]
It would be great ! We also still need this !

**Comment by sauravkedia10** [comment]
If you can also add candlestick charts, it will make a perfect tool. In any cases, for time series aggregated data, we may want to look at many attributes during the aggregation periods, not just the mean/last. In stock market data, we generally want to look at Open. High, Low, Close. They are useful in any time aggregated data-series.

**Comment by torkelo** [comment]
We are trying limit open GitHub issues in order to better track planned work or features we eventually hope to implement rather then enumerate all possible features. 

I will close this issue since it's been open for very long time without much interest in the feature or anyone showing interest in implementing it. 

This does not mean that we will never ever implement it or that we will never accept a PR for it. A closed issue can still attract up votes and act as a ticket to track feature demand/interest.

**Comment by girishjk** [comment]
Not for this specific purpose, but I have only weekdays data, so having a way to exclude weekends will make these graphs look prettier!

**Comment by N3oXid** [comment]
I'm also looking for a way to filter out both weekends and "non business hours".

**Comment by confterm** [comment]
Would be really great to have such filter for the time range.

**Comment by srajangarg** [comment]
Hi, @torkelo any plans to work on this feature? I could also try and implement it myself and submit in a PR if you could point me to the relevant portion of the code.

**Comment by jleskovar-tyro** [comment]
Hi, @torkelo , @srajangarg , also really keen to see this feature, and happy to help.

**Comment by jleskovar-tyro** [comment]
> If you can also add candlestick charts, it will make a perfect tool. In any cases, for time series aggregated data, we may want to look at many attributes during the aggregation periods, not just the mean/last. In stock market data, we generally want to look at Open. High, Low, Close. They are useful in any time aggregated data-series.

There is a (currently unmaintained) Candlestick plugin @ https://github.com/ilgizar/ilgizar-candlestick-panel

**Comment by bjwxh** [comment]
Would like to see it implemented!

**Comment by Scott-St** [comment]
+1 for filtering of timeframes.

**Comment by akrus** [comment]
Would be nice to have this feature.

**Comment by preynoldssco** [comment]
Also interested in this feature. We are tracking equipment use in a University research facility with Grafana - nothing much happens on the weekend so it'd be great to hide those ticks on x axis

**Comment by StefanNeubert** [comment]
+1 for time range exclude

**Comment by jumblies** [comment]
+1

**Comment by LeoWu1993** [comment]
+1

**Comment by debaser19** [comment]
Hi there, I am also interested in this feature for similar purposes. I have a custom setup using Dash and Plotly to plot my graph, and am able to format the ticks and rangebreak like so

```python
fig.update_xaxes(
        tickformat = '%I:%M %p\n%x',
        rangebreaks = [
            dict(bounds = ['sat', 'mon']),
            dict(bounds = [20, 4], pattern = 'hour')
        ]
    )
```

This formats the x axis to only show values M-F 4:00am - 8:00pm. I'd love to have a similar feature with Grafana to be able to do so

**Comment by gaardiolor** [comment]
I'm also interested, would be a useful feature when monitoring solar panels.

**Comment by nawaz2013** [comment]
+1 for this feature of excluding weekends from X axis in grafana

**Comment by gguille077** [comment]
this would be a very nice to have still

**Comment by Tonio101** [comment]
was this implemented?

**Comment by seanlaff** [comment]
I don't believe there is a workaround for this. Here's how weekends look with the candlestick viz

![image](https://user-images.githubusercontent.com/3194832/232353016-f672a2ac-7437-4482-a743-fe5a8c932eb5.png)

**Comment by sweihub** [comment]
Any progress, how to hide the non trading hours?

**Comment by caelan-schneider** [comment]
+1

**Comment by west0r** [comment]
Нou can't add this requested feature for 9 years. What's whrong with you, Grafana team?

**Comment by ysaipranav** [comment]
+1

**Comment by m20xx20** [comment]
We also need to solve this issue.

The Time series panel always renders a continuous time axis. For periods with no data, the available options are:

  1. Disconnect values (Connect values: Never) — a gap appears in the line, but the time axis still occupies the full interval
  2. Connect through nulls (Connect values: Always) — a straight line bridges the gap, spanning across the empty time range

  In both cases the X-axis remains continuous with no way to skip a specified time range. When the inactive period is long, the actual data gets compressed into a small portion of the chart, hurting readability.

**Comment by benorgera** [comment]
I work on a trading system with data for only 8 hours a day on weekdays. It is 2026 now and still, this time series visualization tool requires the assumption that your series is 24/7 for a good UX.

How can the grafana community live with this state of affairs? Googling "grafana exclude weekends" yields countless people asking for this year over year. Now that Claude can write this feature in single digit hours, this is even more embarrassing

**Comment by leeoniya** [comment]
these comments are unproductive.

in the past 2+ years, has anyone actually tried to engage with the complexity outlined in https://github.com/grafana/grafana/issues/2418#issuecomment-2003711997? nope.

> Now that Claude can write this feature in single digit hours

even if it _can_ do it, thoroughly enough to address all the above concerns (and more), Claude is not going to be on the hook for understanding and maintaining the additional extra code for the next 10 years.

maybe you don't use relative time ranges
maybe you don't use cursor sync
maybe you don't use time shift
maybe you don't use time comparison features
maybe you don't need multiple data sources
maybe you don't need different time zones
maybe you don't need multiple x axes
...

except all of these things (and many more) exist, and all need to keep working with this "skip weekends" feature.

> this is even more embarrassing

what's embarrassing is waving away all of this and telling us that it's "easy", having contributed nothing to this project.
